### PR TITLE
EVG-2565 Add Ability to Approve External Pull Request Patches

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: public/static/js/patch_new.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ mciModule.controller('PatchController', function($scope, $filter, $window, notif
   // in the left panel. Also accounts for special toggle behavior when the user
   // is holding shift/meta/ctrl when clicking.
   $scope.selectVariant = function($event, index){
+    $scope.variant = $scope.variants[index]
```
Comment (john-m-liu) [inline public/static/js/patch_new.js]: Do you do any sort of binding to this? If not I would use a smaller scope for this

File: public/static/js/patch_new.js [hunk 2]
```diff
@@ -91,30 +92,37 @@ mciModule.controller('PatchController', function($scope, $filter, $window, notif
     }
   }
 
+  $scope.isUnauthorizedPRPatch = (patch.Version.length === 0 && patch.GithubPatchData.PRNumber !== 0);
+  $scope.isPRPatch = patch.GithubPatchData.PRNumber !== 0;
+
+  $scope.collectVariantTasks = function() {
```
Comment (john-m-liu) [inline public/static/js/patch_new.js]: I don't see that this is called outside of the control so I wouldn't put this in $scope

File: public/static/js/patch_new.js [hunk 3]
```diff
@@ -240,4 +248,22 @@ mciModule.controller('PatchController', function($scope, $filter, $window, notif
...
+
+  $scope.defaultVTs = $scope.collectVariantTasks();
+
+  // When creating authorizing github patches, do not allow the authorizer to
+  // disable the default set of tasks
+  $scope.checkDisabledState = function(task) {
+    if (!$scope.isUnauthorizedPRPatch) {
+        return false;
+    }
+    var variant = $scope.variant.id;
```
Comment (john-m-liu) [inline public/static/js/patch_new.js]: This isn't used

**Comment by richardsamuels** [comment]
@john-m-liu  addressed your comments

**Pull request merged**